datafusion/physical-expr/src/aggregate.rs

```diff
@@ -97,6 +97,165 @@ impl AggregateExprBuilder {
     /// Constructs an `AggregateFunctionExpr` from the builder
     ///
     /// Note that an [`Self::alias`] must be provided before calling this method.
+    ///
+    /// # Example: Create an [`AggregateUDF`]
+    ///
+    /// In the following example, [`AggregateFunctionExpr`] will be built using [`AggregateExprBuilder`]
+    /// which provides a build function. Full example could be accessed from the source file.
+    ///
+    /// ```
+    /// # use std::any::Any;
+    /// # use std::sync::OnceLock;
+    /// # use std::sync::Arc;
+    /// # use arrow::datatypes::DataType;
+    /// # use datafusion_common::{DataFusionError, plan_err, Result, ScalarValue};
+    /// # use datafusion_expr::{col, ColumnarValue, Signature, Volatility, Expr, Documentation};
+    /// # use datafusion_expr::{AggregateUDFImpl, AggregateUDF, Accumulator, function::{AccumulatorArgs, StateFieldsArgs}};
+    /// # use datafusion_expr::window_doc_sections::DOC_SECTION_AGGREGATE;
+    /// # use arrow::datatypes::Schema;
+    /// # use arrow::datatypes::Field;
+    /// # use arrow::array::Array;
+    /// #
+    /// # #[derive(Debug)]
+    /// # struct FirstValueAccumulator {
+    /// #     value: Option<ScalarValue>,
+    /// #     data_type: DataType,
+    /// # }
+    /// #
+    /// # impl Accumulator for FirstValueAccumulator {
+    /// #     fn update_batch(&mut self, values: &[Arc<dyn Array>]) -> Result<()> {
```

Comment:
I think we can significantly shorten the example by just using `unimplemented()!` instead of adding an actual implementation for an accumulator, and the methods.